main.tf

```diff
@@ -478,6 +492,15 @@ resource "azurerm_windows_virtual_machine" "vm_windows" {
       condition     = var.network_interface_ids != null || var.new_network_interface != null
       error_message = "Either `new_network_interface` or `network_interface_ids` must be provided."
     }
+    precondition {
+      condition     = var.bypass_platform_safety_checks_on_user_schedule_enabled == false ? true : local.patch_mode == "AutomaticByPlatform"
```

Comment:
If `var.bypass_platform_safety_checks_on_user_schedule_enabled` cannot be `null`, then this expression could be simplify to:

```hcl
condition     = !var.bypass_platform_safety_checks_on_user_schedule_enabled || local.patch_mode == "AutomaticByPlatform"
```